.github/workflows/pytest.yaml

```diff
@@ -37,12 +37,12 @@ jobs:
       uses: actions/cache@v3.0.2
       with:
         path: ${{ steps.prepare-cache-pip.outputs.cache-dir }}
-        key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('**/test_requirements.txt') }}-${{ hashFiles('**/setup.py') }}
+        key: ${{ runner.os }}-${{ matrix.python-version }}-pip-${{ hashFiles('**/requirements.txt') }}-${{ hashFiles('**/setup.cfg') }}
         restore-keys: |
           ${{ runner.os }}-${{ matrix.python-version }}-pip-
-    - name: install test_requirements.txt
+    - name: install requirements.txt
       run: |
-        pip install -r test_requirements.txt
+        pip install -r requirements.txt
```

Comment:
What is the logic between test_requirements, docs_requrements and requirements expected to be?

Reply:
@jenshnielsen we have had that conversation some time ago. The way I remember the conversation was that:

- we don't use `requirements.txt`
- `setup.cfg` has the dependencies as loosely marked as possible, and offers different sets for baisc install, test and docs
- `test_requirements.txt` and `docs_requirements.txt` pins the requirements to an exact version for running in CI (or reproducing locally.

We also talked about alternatively:
Don't use any requirement files. Use `setup.cfg` with a `test` and `docs` that pin the requirements for those.

I would be in favor of the last one as it slashes down book keeping and if someone wants to run tests locally, it is okay to have to create a new env for this as we probably want to reproduce the results as in CI.

Reply:
> we don't use requirements.txt

..except when dependabot is used, e.g. qcodes.

----

i'd say that it's a choice between keeping things consistent with qcodes or not. and i am in favor of keeping things consistent.

~~in addition, my unerstanding of the concepts tell me that "requirements" files work well as "lock files", while install-requires and extra-requires in setup.cfg should not be 100% pinned since they are reuqirements for ":running" the package (and [test] is for "running" tests since they are now shipped with the package ;) ).~~ don't want to have a discussion now, so ignore this additional comment, make a decision and let's go with it :)

Reply:
ok, thank you two for the learning @astafan8 @jenshnielsen !

Reply:
In the spirit of 2) would it then be an option to not specify test- and docs- requirements in setup.cfg at all? (and move docs and test_requirements 1:1 to setup.cfg once depandabot works with it) I am not sure about the use of the 'unpinned' test dependencies.

Reply:
> to not specify test- and docs- requirements in setup.cfg at all

yes. except, it's nice to specify test so that we can install the package with [test] in order to run the tests (not in CI) because they get shipped with the package now

Reply:
@jenshnielsen, @astafan8, @Dominik-Vogel I thought this might be debated 😁!

>What is the logic between test_requirements, docs_requrements and requirements expected to be?

I went down a bit of a rabbit whole to around this question and I found the [PEP665 discourse discussion](https://discuss.python.org/t/pep-665-specifying-installation-requirements-for-python-projects/9911) to be helpful even though it was rejected.  This following comment highlights how universal and still not really solved this issue is in the python community.  
>PEP 621 also (purposefully) omits two key details that often become important for projects:
>- A listing of all indirect/transitive dependencies
>- Specifying (at least) specific versions of dependencies for reproducible installations

>don't want to have a discussion now, so ignore this additional comment, make a decision and let's go with it :)

Sorry @astafan8 apparently I wanted to have this discussion now (read your dashed lines after I wrote all this 😆) but I agree with your statement.

Also It looks like there is a bit of activity on this topic https://github.com/pypa/pip/pull/10748 as an example which is trying to address:
>With this change, users can use pip's powerful new resolver to extract metadata from a package resolution, such as what dependencies each package requires. This is currently not possible in pip or any other tool I am aware of.
>
>Getting more specific, this result can then be processed by external tools to:
>- generate a lockfile.
>- check whether transitive dependencies would introduce a conflict.
> - download packages directly from download URLs without having to traverse PyPI again.

For now, my feelings are that we should use a single requirements file generated from pip freeze or pip-tools pip-compile to pin the dependencies that will be used to build broadbean for tests and build docs in CI. To remove confusion we could place the requirments.txt in .github/static/ as well. 

From a user perspective the only install will be from setup.cfg or a released package. This would mean we'd need to keep setup.cfg in sync with any issues dependabot has raised by the requirments.txt bumps. Hopefully we'll see a cleaner solution soon but this is the best I could put together.

What do you all think? 

>We also talked about alternatively:
Don't use any requirement files. Use setup.cfg with a test and docs that pin the requirements for those.

Did we come to a consensus on if we plan to use depdendabot on this project?  W

Reply:
@trevormorgan sounds fine, allow me to repeat your suggestion but make it more explicit and please correct it if i got something worng:
* create with pip freeze a single requirements.txt file that will include ALSO tests and docs requirements, strictly pinned; and then use this requirements.txt file in CI
* keep setup.cfg with open dependency specification (e.g. like now)
* add a [test] section to setup.cfg with relatively open dependency specification so that users can run tests locally when they install the package

correct?

Reply:
@astafan8 thanks for summarizing and yes that's correct. I'll get these changes in so we can merge this PR.